tests/unit/plugins/modules/test_pacemaker_resource.yaml

```
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) Dexter Le (dextersydney2001@gmail.com)
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+---
+anchors:
+  environ: &env-def {environ_update: {LANGUAGE: C, LC_ALL: C}, check_rc: false}
+test_cases:
+- id: test_missing_input
+  input: {}
+  output:
+    failed: true
+    msg: "missing required arguments: name"
+- id: test_present_minimal_input_resource_not_exist
+  input:
+    state: present
+    name: virtual-ip
+    resource_type:
+      resource_name: IPaddr2
+    resource_option:
+      - "ip=[192.168.999.5]"
+  output:
+    changed: true
+    previous_value:
+    value: "  * virtual-ip\t(ocf:heartbeat:IPAddr2):\t Started"
+  mocks:
+    run_command:
+    - command: ["/testbin/pcs", resource, status, virtual-ip]
+      environ: *env-def
+      rc: 1
+      out: ""
+      err: ""
+    - command: ["/testbin/pcs", resource, create, virtual-ip, IPaddr2, "ip=[192.168.999.5]", "--wait=300"]
+      environ: *env-def
+      rc: 0
+      out: ""
+      err: ""
+    - command: ["/testbin/pcs", resource, status, virtual-ip]
+      environ: *env-def
+      rc: 0
+      out: "  * virtual-ip\t(ocf:heartbeat:IPAddr2):\t Started"
+      err: ""
+- id: test_present_minimal_input_resource_exists
+  input:
+    state: present
+    name: virtual-ip
+    resource_type:
+      resource_name: IPaddr2
+    resource_option:
+      - "ip=[192.168.999.5]"
+  output:
+    changed: false
+    previous_value: "  * virtual-ip\t(ocf:heartbeat:IPAddr2):\t Started"
+    value: "  * virtual-ip\t(ocf:heartbeat:IPAddr2):\t Started"
+  mocks:
+    run_command:
+    - command: ["/testbin/pcs", resource, status, virtual-ip]
+      environ: *env-def
+      rc: 0
+      out: "  * virtual-ip\t(ocf:heartbeat:IPAddr2):\t Started"
+      err: ""
+    - command: ["/testbin/pcs", resource, create, virtual-ip, IPaddr2, "ip=[192.168.999.5]", "--wait=300"]
+      environ: *env-def
+      rc: 1
+      out: ""
+      err: "Error: 'virtual-ip' already exists\n"
+    - command: ["/testbin/pcs", resource, status, virtual-ip]
+      environ: *env-def
+      rc: 0
+      out: "  * virtual-ip\t(ocf:heartbeat:IPAddr2):\t Started"
+      err: ""
+- id: test_absent_minimal_input_resource_not_exist
+  input:
+    state: absent
+    name: virtual-ip
+  output:
+    changed: false
+    previous_value:
+    value:
+  mocks:
+    run_command:
+    - command: ["/testbin/pcs", resource, status, virtual-ip]
+      environ: *env-def
+      rc: 1
+      out: ""
+      err: ""
+    - command: ["/testbin/pcs", resource, remove, virtual-ip]
+      environ: *env-def
+      rc: 1
+      out: ""
+      err: "Error: Resource 'virtual-ip' does not exist.\n"
+- id: test_absent_minimal_input_resource_exists
+  input:
+    state: absent
+    name: virtual-ip
+  output:
+    changed: true
+    previous_value: "  * virtual-ip\t(ocf:heartbeat:IPAddr2):\t Started"
+    value:
```

Not sure what's the proper way to handle this? If I specify `null` the helper gives error of assertion error that 'None' != 'None'. Leaving this empty allows the unit test to pass.

---

Ha. Good catch. I need to fix that in the test helper. The test should be `is None` instead of `== None`. I shall raise a bug report for that in test helper.

---

err, I am trying to reproduce the issue but adding `null` to the YAML caused no error to me. :astonished: could you please add it and let the error show here in the CI?
```suggestion
    value: null
```

---

It might be specific to some combination of ansible core version and python version, I'd rather let the CI expose it than try to spot that needle in the haystack locally.